pkg/proxy/proxy.go

```diff
@@ -346,7 +340,7 @@ func (p *Proxy) removeRedirect(id string) {
 
 	// Delay the release and reuse of the port number so it is guaranteed to be
 	// safe to listen on the port again.
-	proxyPort := r.listener.ProxyPort
+	proxyPort := r.proxyPort.ProxyPort
```

Comment:
nit: now that `listener` got renamed to `proxyPort` - it might makes sense to rename `ProxyPort` to `PortNumber`?

Reply:
We generally use ProxyPort explicitly, since port number is ambiguous -- it could be the real destination port (e.g. 443) or Envoy's port (random)